pkg/controller/common/util.go

```diff
@@ -1473,3 +1474,72 @@ func UpdateImageIOAnnotations(annotations map[string]string, imageio *cdiv1.Data
 	annotations[AnnCertConfigMap] = imageio.CertConfigMap
 	annotations[AnnDiskID] = imageio.DiskID
 }
+
+// GetSnapshotClassForSmartClone looks up the snapshot class based on the storage class
+func GetSnapshotClassForSmartClone(dvName string, targetPvcStorageClassName *string, log logr.Logger, client client.Client) (string, error) {
+	logger := log.WithName("GetSnapshotClassForSmartClone").V(3)
+	// Check if relevant CRDs are available
+	if !isCsiCrdsDeployed(client, log) {
+		logger.Info("Missing CSI snapshotter CRDs, falling back to host assisted clone")
+		return "", nil
+	}
+
+	targetStorageClass, err := GetStorageClassByName(client, targetPvcStorageClassName)
+	if err != nil {
+		return "", err
+	}
+	if targetStorageClass == nil {
+		logger.Info("Target PVC's Storage Class not found")
+		return "", nil
+	}
+
```

Comment:
any clue why there was another `Fetch the source storage class` before the func moved here?

Reply:
guess a leftover from the refactoring of https://github.com/kubevirt/containerized-data-importer/blob/release-v1.34/pkg/controller/datavolume-controller.go#L1160